engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schema/StringForcingDatumReader.scala

```diff
@@ -0,0 +1,47 @@
+package pl.touk.nussknacker.engine.avro.schema
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.avro.Schema
+import org.apache.avro.generic.{GenericData, GenericDatumReader}
+import org.apache.avro.io.Decoder
+import org.apache.avro.reflect.{ReflectData, ReflectDatumReader}
+import org.apache.avro.specific.{SpecificData, SpecificDatumReader}
+import org.apache.avro.util.Utf8
+
+import scala.util.{Properties, Try}
+
+trait StringForcingDatumReader[T] extends GenericDatumReader[T]  {
+  self: GenericDatumReader[T] =>
+
+  override def readString(old: Any, expected: Schema, in: Decoder): AnyRef = super.readString(old, expected, in) match {
+    case data: Utf8 if AvroStringSettings.forceUsingStringForStringSchema => data.toString
+    case data => data
+  }
+
+  override def findStringClass(schema: Schema): Class[_] = {
+    if (schema.getType == Schema.Type.STRING && AvroStringSettings.forceUsingStringForStringSchema) classOf[String]
+    else super.findStringClass(schema)
+  }
+}
+
+object AvroStringSettings extends LazyLogging {
+  val default: Boolean = false
+  val envName = "AVRO_USE_STRING_FOR_STRING_TYPE"
+
+  lazy val forceUsingStringForStringSchema: Boolean = Properties.envOrNone(envName)
+    .map(str => Try(str.toBoolean).getOrElse {
+      logger.warn(s"$envName is not valid boolean value. Using default $default value")
+      default
+    }).getOrElse(default)
+}
+
+object StringForcingDatumReader {
+  def forGenericDatumReader[T](writer: Schema, reader: Schema, data: GenericData): GenericDatumReader[T] =
```

Comment:
Maybe we can just skip "for" in name? Because "string forcing datum reader for generic datum reader" looks weird. Maybe just `StringForcingDatumReader.genericDatumReader`?

Reply:
you're rigth it really sound wird - changed